File: Day9/ArmStrongNumber.java
import java.util.Scanner;
class ArmStrongNumber {

    public static int getDigitCount(int num){

        int count = 0;
        while(num > 0){
         count = count +1;
         num = num / 10;
        }
        return count;
    }
    public static int powerOf(int digit ,int power){

       int prod = 1;
       while(power>0){
         prod = prod * digit;
         power--;
       }
       return prod;
    }  
    public static boolean getIsArmStrongNumber(int num){
        int temp = num;
        int sum= 0;
        int power = getDigitCount(num);
        while(num != 0){
            int  digit = num %10;
            sum = sum + powerOf(digit,power);
            num = num / 10;
        }
        return temp == sum;

    }
    public static void main(String args[]){       

        Scanner s1 = new Scanner(System.in);
        System.out.println(" Enter the Number : ");
        int num =s1.nextInt();
        s1.close();
        boolean res = getIsArmStrongNumber(num);
        if (res == true){
            System.out.println(num + " is a ArmStrong Number ");
        }else{
            System.out.println(num + " is not a ArmStrong Number");
        }   
    }  
}

File: Day9/armstrong_number.py
""" Program to check number is Arstrong or not."""
#function 1
def get_digit_count(n):
    """ Calculates the number of digits in a given number. """
    count = 0
    temp = n
    while temp>0:
        count += 1
        temp //=  10
    return count

#function 2 to get the power of the digit
def power_of(digit,power):
    """
    Calculates the power of a given digit.
    """
    prod=1 
    while power>0:
        prod *= digit
        power -= 1
    return prod
#function 3 to check the number is ArmStrong or not
def is_armstrong_number(n):
    """Checks if a given number is an Armstrong number.
    """
    temp=n
    total_sum = 0
    power=get_digit_count(num)
    while temp>0:
        digit = temp % 10
        total_sum +=  power_of(digit,power)
        temp //= 10
    return num == total_sum

num = int(input("Enter the number : "))
ans = is_armstrong_number(num)
if ans:
    print(num,"is a Arm Strong Number. ")
else:
    print(num,"is not a Arm Strong Number.")
